[BUG] Unable to build on OpenBSD

## Description

When trying the `pavao` crate on OpenBSD current/amd64 (future version 7.6), I have some errors in `src/smb/types/stat.rs` file.

## Steps to reproduce

- OpenBSD current/amd64 with Rust 1.79.0
- Samba librairies installed with OpenBSD `samba` package
-  Clone `pavao` sources via `git clone`
- Build crate with `cargo build`
```shell
$ cargo build -v
(...)
     Running `rustc --crate-name pavao --edition=2021 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=274 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=62bcaf34a118df01 -C extra-filename=-62bcaf34a118df01 --out-dir /home/fox/dev/rust-pavao.git/target/debug/deps -C incremental=/home/fox/dev/rust-pavao.git/target/debug/incremental -L dependency=/home/fox/dev/rust-pavao.git/target/debug/deps --extern lazy_static=/home/fox/dev/rust-pavao.git/target/debug/deps/liblazy_static-7d8bc0f4ae642bf6.rmeta --extern libc=/home/fox/dev/rust-pavao.git/target/debug/deps/liblibc-97aace3d6bb24e94.rmeta --extern log=/home/fox/dev/rust-pavao.git/target/debug/deps/liblog-84f9b4f84b36243a.rmeta --extern thiserror=/home/fox/dev/rust-pavao.git/target/debug/deps/libthiserror-62a9f249e6c4ede4.rmeta -L native=/usr/local/lib -l smbclient -l smbclient --cfg x86_64 --cfg unix`
error[E0063]: missing fields `bavail`, `bfree`, `blocks` and 3 other fields in initializer of `types::stat::SmbStatVfs`
  --> src/smb/types/stat.rs:47:9
   |
47 |         Self {
   |         ^^^^ missing `bavail`, `bfree`, `blocks` and 3 other fields

error[E0063]: missing fields `blksize`, `dev`, `nlink` and 1 other field in initializer of `types::stat::SmbStat`
   --> src/smb/types/stat.rs:147:9
    |
147 |         Self {
    |         ^^^^ missing `blksize`, `dev`, `nlink` and 1 other field

For more information about this error, try `rustc --explain E0063`.
error: could not compile `pavao` (lib) due to 2 previous errors
```

In `src/smb/types/stat.rs` file, the `SmbStatVfs` and `SmbStat` structs are not implemented for other OS than macOS and Linux (use of `cfg` keyword for conditional compilation).

--- a/README.md
+++ b/README.md
@@ -76,6 +76,7 @@
       - [MacOS 🍎](#macos-)
       - [Debian based systems 🐧](#debian-based-systems-)
       - [RedHat based systems 🐧](#redhat-based-systems-)
+      - [OpenBSD 🐡](#openbsd-)
       - [Build from sources 📁](#build-from-sources-)
     - [Create a pavao application](#create-a-pavao-application)
     - [Run examples](#run-examples)
@@ -136,6 +137,14 @@ dnf install libsmbclient-devel libsmbclient
 ```
 
 ⚠️ `libsmbclient-devel` is required only on the machine where you build the application
+
+#### OpenBSD 🐡
+
+Install samba with pkg_add:
+
+```sh
+pkg_add samba
+```
 
 #### Build from sources 📁
 

--- a/src/smb/types/stat.rs
+++ b/src/smb/types/stat.rs
@@ -57,6 +57,8 @@ impl From<statvfs> for SmbStatVfs {
             blocks: s.f_blocks as u64,
             #[cfg(linux_riscv64)]
             blocks: s.f_blocks,
+            #[cfg(target_os = "openbsd")]
+            blocks: s.f_blocks,
             #[cfg(target_os = "macos")]
             bfree: s.f_bfree as u64,
             #[cfg(linux_x86_64)]
@@ -66,6 +68,8 @@ impl From<statvfs> for SmbStatVfs {
             #[cfg(linux_arm)]
             bfree: s.f_bfree as u64,
             #[cfg(linux_riscv64)]
+            bfree: s.f_bfree,
+            #[cfg(target_os = "openbsd")]
             bfree: s.f_bfree,
             #[cfg(target_os = "macos")]
             bavail: s.f_bavail as u64,
@@ -77,6 +81,8 @@ impl From<statvfs> for SmbStatVfs {
             bavail: s.f_bavail as u64,
             #[cfg(linux_riscv64)]
             bavail: s.f_bavail,
+            #[cfg(target_os = "openbsd")]
+            bavail: s.f_bavail,
             #[cfg(target_os = "macos")]
             files: s.f_files as u64,
             #[cfg(linux_x86_64)]
@@ -86,6 +92,8 @@ impl From<statvfs> for SmbStatVfs {
             #[cfg(linux_arm)]
             files: s.f_files as u64,
             #[cfg(linux_riscv64)]
+            files: s.f_files,
+            #[cfg(target_os = "openbsd")]
             files: s.f_files,
             #[cfg(target_os = "macos")]
             ffree: s.f_ffree as u64,
@@ -97,6 +105,8 @@ impl From<statvfs> for SmbStatVfs {
             ffree: s.f_ffree as u64,
             #[cfg(linux_riscv64)]
             ffree: s.f_ffree,
+            #[cfg(target_os = "openbsd")]
+            ffree: s.f_ffree,
             #[cfg(target_os = "macos")]
             favail: s.f_favail as u64,
             #[cfg(linux_x86_64)]
@@ -106,6 +116,8 @@ impl From<statvfs> for SmbStatVfs {
             #[cfg(linux_arm)]
             favail: s.f_favail as u64,
             #[cfg(linux_riscv64)]
+            favail: s.f_favail,
+            #[cfg(target_os = "openbsd")]
             favail: s.f_favail,
             fsid: s.f_fsid as u64,
             flag: s.f_flag as u64,
@@ -158,6 +170,8 @@ impl From<stat> for SmbStat {
             #[cfg(linux_riscv64)]
             blksize: s.st_blksize as i64,
             created: time_t_to_system_time(s.st_ctime),
+            #[cfg(target_os = "openbsd")]
+            blksize: s.st_blksize as i64,
             #[cfg(target_os = "macos")]
             dev: s.st_dev,
             #[cfg(linux_x86_64)]
@@ -171,6 +185,8 @@ impl From<stat> for SmbStat {
             gid: s.st_gid,
             mode: SmbMode::from(s.st_mode),
             modified: time_t_to_system_time(s.st_mtime),
+            #[cfg(target_os = "openbsd")]
+            dev: s.st_dev as i32,
             #[cfg(target_os = "macos")]
             nlink: s.st_nlink as u64,
             #[cfg(linux_x86_64)]
@@ -181,6 +197,8 @@ impl From<stat> for SmbStat {
             nlink: s.st_nlink as u64,
             #[cfg(linux_riscv64)]
             nlink: s.st_nlink as u64,
+            #[cfg(target_os = "openbsd")]
+            nlink: s.st_nlink as u64,
             #[cfg(target_os = "macos")]
             rdev: s.st_rdev as u64,
             #[cfg(linux_x86_64)]
@@ -190,6 +208,8 @@ impl From<stat> for SmbStat {
             #[cfg(linux_arm)]
             rdev: s.st_rdev as u64,
             #[cfg(linux_riscv64)]
+            rdev: s.st_rdev as u64,
+            #[cfg(target_os = "openbsd")]
             rdev: s.st_rdev as u64,
             size: s.st_size as u64,
             uid: s.st_uid,

